src/cli/commands/test/mod.rs

```diff
@@ -0,0 +1,164 @@
+
+#[derive(Args, Debug)]
+pub struct TestArgs {
+	/// Path to a cairo directory
+	#[clap(short, long, value_hint=ValueHint::DirPath, value_parser=path_is_valid_directory)]
+	pub root: PathBuf,
+
+
+
+
+}
+
+fn list_test_entrypoints(compiled_path: &PathBuf) -> Result<Vec<String>, String> {
+	let re = Regex::new(r"__main__.(test_\w+)").map_err(|_op| format!("Regex failed"))?;
+	let data = fs::read_to_string(compiled_path).map_err(|_op| format!("File does not exist"))?;
+	let json = serde_json::from_str::<Value>(&data).map_err(|_op|format!("Compile fails: {}", compiled_path.display()))?;
+	let mut test_entrypoints = Vec::new();
+	// if json is empty, throw error
+	if json.is_null() {
+		return Err(format!("Json file is empty"));
+	}
+	for (key, value) in json["identifiers"].as_object().unwrap() {
+		if re.is_match(key) && value["type"] == "function" {	
+			for capture in re.captures_iter(key) {
+				test_entrypoints.push(capture[1].to_string());
+			}
+		}
+	}
+
+	Ok(test_entrypoints)
+}
+
+
+/// Execute command output
+#[derive(Debug, Serialize)]
+pub struct TestOutput(Vec<u8>);
+
+impl Write for TestOutput {
+	fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+		self.0.write(buf)
+	}
+
+	fn flush(&mut self) -> std::io::Result<()> {
+		self.0.flush()
+	}
+}
+
+
+
+impl Display for TestOutput {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(
+			f,
+			"{}",
+			from_utf8(&self.0).map_err(|e| {
+				error!("failed to format the execution output due to invalid utf8 encodig: {e}");
+				std::fmt::Error
+			})?
+		)
+	}
+}
+
+
+
+impl CommandExecution<TestOutput> for TestArgs {
+	fn exec(&self) -> Result<TestOutput, String> {
+		let hint = HintFunc(Box::new(greater_than_hint));
+		let mut hint_processor = BuiltinHintProcessor::new_empty();
+		hint_processor.add_hint(String::from("print(ids.a > ids.b)"), hint);
+
+		let result = ListArgs {
+			root: PathBuf::from(&self.root),
+		}.exec();
+		
+		let mut output = TestOutput(vec![]);
+		for file in result?.files {
+			let compiled_program_path = compile(&file);
+			// Check if the compilation was successful
+			let compiled_program_path = match compiled_program_path {
+				Ok(res) => res,
+				Err(e) => {
+					eprintln!("{}", e);
+					continue;
+				},
+			};
+			let test_entrypoints = list_test_entrypoints(&compiled_program_path);
+
+			let test_entrypoints = match test_entrypoints {
+				Ok(res) => res,
+				Err(e) => {
+					eprintln!("{}", e);
+					continue;
+				}
+			};
+
+			if test_entrypoints.is_empty() {
+				println!("No test entrypoints found in {}", compiled_program_path.display());
+			}
+			
+			for test_entrypoint in test_entrypoints {
+				let cairo_runner = cairo_run(&compiled_program_path, &test_entrypoint, false, &hint_processor);
+				let mut result = match cairo_runner {
+				Ok(res) => res,
+				Err(_) => {
+					eprintln!("failed to run test: {}", test_entrypoint);
+					continue;
+				}
+			};
```

Comment:
That's not what we want to do.
We want to either print Test Pass or Test Failed to the `exec` `output`